keps/sig-multicluster/5339-clusterprofile-plugin-credentials/README.md

```diff
@@ -0,0 +1,619 @@
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+To manage an Inventory of Clusters, a platform admin can rely on having the cluster manager
+output ClusterProfile CRs that point to the clusters. Those CRs are key for multicluster controllers
+that want to operate on the clusters. However, there isn't a single way to obtain credentials
+to reach those clusters. This KEP provides a standardized way to obtain credentials for Clusters
+when using ClusterProfile and makes it pluggable to allow the diverse ecosystem to support the
+multitude of ways to obtain credentials. It also reuses part of the Kubeconfig external provider
+semantics to make implementation easier.
+
+## Motivation
+
+ClusterInventory is unfinished without an ability to use the clusters and controller
+writers have been very explicit that credentials are needed. Previous attempts at writing credentials have
+failed and we believe that a plugin model, also reusing known flows, will help solve the "credentials" need
+for ClusterProfiles.
+
+See [introduction slides](https://docs.google.com/presentation/d/1v5-J-kFJ3TSpKqSraHcYkCz2NG7cNnYpq0ISF85wNMU/edit)
+
+### Goals
+
+* Provide a library for controllers to obtain credentials for a cluster represented by a ClusterProfile
+* Allow cluster managers to provide a method to obtain credentials that doesn't require to be embedded into the controller code
+  and recompiling.
+* Be a secure mechanism for credential obtention and storage.
```

Comment:
I guess it would not hurt to get a buy-in from the sig Auth

Reply:
I agree, could you invite a couple people to this KEP? I don't remember POCs.
Given how similar it is to the kubeconfig exec, I do expect good inputs from them.